graphql_compiler/schema_generation/schema_graph.py

```diff
@@ -509,84 +471,56 @@ def _set_up_schema_elements_of_kind(self, class_name_to_definition, kind, class_
+                if property_name in link_direction_to_endpoint_classes:
+                    self._validate_link(orientdb_property_definition, class_name,
+                                        class_name_to_definition)
+                    link_direction_to_endpoint_classes[property_name].append(
+                        orientdb_property_definition['linkedClass'])
                 else:
+                    property_descriptor = self._create_descriptor_from_orientdb_property_definition(
+                        class_name, orientdb_property_definition)
                     property_name_to_descriptor[property_name] = property_descriptor
 
-            for property_name in allowed_duplicated_edge_property_names:
-                elements = {
-                    property_descriptor.qualifier
-                    for property_descriptor in links[property_name]
-                }
-                # If there are multiple in/out properties, we choose to include the one that
-                # is a subclass of all the elements present in the in/out properties.
-                for property_descriptor in links[property_name]:
-                    subclass_set = self._subclass_sets[property_descriptor.qualifier]
-                    if len(elements.intersection(subclass_set)) == 1:
-                        current_descriptor = property_name_to_descriptor.get(property_name, None)
-                        if current_descriptor and current_descriptor != property_descriptor:
-                            raise AssertionError(u'There already exists property "{}" in addition '
-                                                 u'to property "{}" which is a subclass of all '
-                                                 u'in/out properties for class "{}".'
-                                                 .format(current_descriptor,
-                                                         property_descriptor, class_name))
-                        property_name_to_descriptor[property_name] = property_descriptor
-
-                if (property_name not in property_name_to_descriptor and not abstract and
-                        kind == SchemaElement.ELEMENT_KIND_EDGE):
-                    raise AssertionError(u'For property "{}" of non-abstract edge class "{}", '
-                                         u'no such subclass-of-all-elements exists.'
-                                         .format(property_name, class_name))
-
             self._elements[class_name] = SchemaElement(class_name, kind, abstract,
                                                        property_name_to_descriptor, class_fields)
 
-    def _create_descriptor_from_property_definition(self, class_name, property_definition,
-                                                    class_name_to_definition):
+            if kind == SchemaElement.ELEMENT_KIND_EDGE:
+                in_leaf_endpoint = self._try_get_base_endpoint(class_name,
```

Comment:
Consider starting the arguments to the function on a new line with one indent. The visual indent at the `(` like this produces lots of whitespace on the left, and these columns of code on the right that at least I find difficult to read.